### PR TITLE
[nrf toup] ci: enable windows 'sanitycheck --build-only'

### PR DESCRIPTION
**Pull request by thst-nordic**
Small changes that enable windows sanitycheck --build-only.
Enables CI to build apps/samples on windows agents.

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -3173,6 +3176,13 @@ def main():
     global run_individual_tests
     options = parse_arguments()
 
+    if (os.name == 'nt'):
+        print("Running sanitycheck on Windows is not supported yet.")
+        print("https://github.com/zephyrproject-rtos/zephyr/issues/2664")
+        if (not options.build_only):
+            print("Running sanitycheck on Windows with '--build-only' might work.")
```
Comment (SebastianBoe) [inline scripts/sanitycheck]: Why only "might work" ?

If this is useful enough to be upstreamed then we would want to do so to avoid conflicts.

Comment (thst-nordic) [reply scripts/sanitycheck]: I tested windows sanitycheck on zephyr samples and it requires one more flag: '--disable-size-report'.   

I've added a check for this in the script.

**Comment by tstilz** [comment]
The Kconfig error from CI is on master already, and was introduced from upmerge on friday.  I fixed half of the problems on PR-160,, but we just have to ignore the others until we can upgrade west.   

It passed all other tests on CI.

**Comment by thst-nordic** [comment]
The Kconfig error from CI is on master already, and was introduced from upmerge on friday. I fixed half of the problems on PR-160,, but we just have to ignore the others until we can upgrade west.

It passed all other tests on CI.

File: scripts/sanitycheck [hunk 2]
```diff
@@ -3208,6 +3211,15 @@ def main():
     global run_individual_tests
     options = parse_arguments()
 
+    if (os.name == 'nt'):
+        print("Running sanitycheck on Windows is not supported yet.")
+        print("https://github.com/zephyrproject-rtos/zephyr/issues/2664")
```
Comment (SebastianBoe) [inline scripts/sanitycheck]: If I'm using a supported operation, like --build-only, then I don't want to be warned that what
I'm doing is not supported.

This should either be removed or merged into the below message.

**Comment by SebastianBoe** [comment]
Please add a link from here to the upstream PR when available.

The upstream PR being created at some point is necessary for this to be merged, but it is not necessary to create it before this is merged.

**Pull request closed without merging**